### PR TITLE
Handle error type arguments in DiagnosticAnalyzerAPIUsageAnalyzer

### LINKED ISSUE #1711: NRE in CSharpDiagnosticAnalyzerApiUsageAnalyzer
See https://ci.dot.net/job/dotnet_roslyn-analyzers/job/master/job/Windows_NT_Release_prtest/5/console

```
12:16:56 CSC : error AD0001: Analyzer 'Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.CSharpDiagnosticAnalyzerApiUsageAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [D:\j\workspace\Windows_NT_Re---c6b3d408\src\Microsoft.CodeAnalysis.Analyzers\Core\Microsoft.CodeAnalysis.Analyzers.csproj]
```

### LINKED ISSUE #1771: Intermittent NullReferenceException in CSharpDiagnosticAnalyzerApiUsageAnalyzer
_From @yaakov-h on August 7, 2018 3:58_

**Version Used**: 
Microsoft.CodeAnalysis.Analyzers v2.6.1

**Steps to Reproduce**:

1. Clone https://github.com/WiseTechGlobal/WTG.Analyzers at commit 223095d2
2. Build from source

**Expected Behavior**:

No null reference exceptions in MS analyzers

**Actual Behavior**:

Sometimes null reference exception in MS analyzers

> CSC : error AD0001: Analyzer 'Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.CSharpDiagnosticAnalyzerApiUsageAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [C:\projects\wtg-analyzers\WTG.Analyzers.Test\WTG.Analyzers.Test.csproj]

_Copied from original issue: dotnet/roslyn#29119_

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #1711
Fixes #1771

### CODE REVIEW

File: src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/DiagnosticAnalyzerApiUsageAnalyzerTests.cs [hunk 1]
```diff
@@ -33,6 +33,7 @@ class MyAnalyzer : DiagnosticAnalyzer
 
     public override void Initialize(AnalysisContext context)
     {
+        var x = typeof(I<>);
```
Comment (mavasani) [inline src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/DiagnosticAnalyzerApiUsageAnalyzerTests.cs]: Verified the NRE in unit test with just the test change.

**Comment by mavasani** [comment]
@dotnet-bot retest prtest/Windows/NT/Debug please

**Comment by mavasani** [comment]
All the intermittent test failures in the repo are happening due to opencover issues:

https://ci.dot.net/job/dotnet_roslyn-analyzers/job/master/job/Windows_NT_Debug_prtest/62/artifact/artifacts/Debug/TestResults/Microsoft.CodeAnalysis.Analyzers.UnitTests_net46_x64.log
```
Failed to register(user:True,register:True,is64:True):5 the profiler assembly; you may want to look into permissions or using the -register:user option instead. C:\Windows\system32\regsvr32.exe /s  /n /i:user "D:\j\workspace\Windows_NT_De---ec62cf09\.packages\opencover\4.6.519\tools\x64\OpenCover.Profiler.dll"
If you are unable to resolve the issue please contact the OpenCover development team
see https://www.github.com/opencover/opencover/issues
```

**Comment by mavasani** [comment]
Tagging @sharwell

**Comment by mavasani** [comment]
@dotnet-bot retest prtest/Windows/NT/Debug please

**Comment by chrischu** [comment]
Is there a ETA for the next release of the NuGet package with that fix included?

**Pull request merged**